src/api/app/models/workflow_run.rb

```diff
@@ -26,8 +26,13 @@ class WorkflowRun < ApplicationRecord
             :workflow_configuration_path, :workflow_configuration_url,
             :hook_event, :hook_action, :generic_event_type,
             :repository_name, :repository_owner, :event_source_name, length: { maximum: 255 }
-  validates :request_headers, :status, :scm_vendor, :hook_event, presence: true
+  validates :request_headers, :status, :scm_vendor, :hook_event, :request_payload, presence: true
   validates :workflow_configuration, length: { maximum: 65_535 }
+  validates :scm_vendor, inclusion: { in: %w[github gitlab gitea], message: "unsupported '%{value}'" }, if: -> { scm_vendor.present? }
+  validates :hook_event, inclusion: { in: ALLOWED_GITHUB_EVENTS, allow_nil: true, message: "unsupported '%{value}'" }, if: -> { scm_vendor == 'github' }
+  validates :hook_event, inclusion: { in: ALLOWED_GITLAB_EVENTS, allow_nil: true, message: "unsupported '%{value}'" }, if: -> { scm_vendor == 'gitlab' }
+  validates :hook_action, inclusion: { in: ALLOWED_PULL_REQUEST_ACTIONS, allow_nil: true, message: "unsupported '%{value}'" }, if: -> { scm_vendor == 'github' }
+  validates :hook_action, inclusion: { in: ALLOWED_MERGE_REQUEST_ACTIONS, allow_nil: true, message: "unsupported event acton '%{value}'" }, if: -> { scm_vendor == 'gitlab' }
```

Comment:
What about gitea?

Reply:
Its already there, I forgot it and added it later. I guess you haven't seen it yet if you go commit by commit. To many commits...